clojure/src/ch01_evaluator.clj

```diff
@@ -0,0 +1,112 @@
+(ns ch01-evaluator
+  "Chapter 1: Basic evaluator.
+  Starts with section 1.2 on p. 4.")
+
+;;; 1.2 (p.4): define `evaluate` signature
+(defn evaluate [exp env])
+
+
+;;; 1.3 (p.4): start building the implementation
+;;; - introduces `atom?` check and `case` for non-atomic forms
+;;; - discusses the difference between 'program' and 'representation';
+;;;   evaluate works on the _representation_ 
+
+;; they define atom as anything that's not a "pair" (cons cell)
+;; - we use `list?` for Clojure which should be good enough
+(def atom? (complement list?))
```

Comment:
I've seen `def` (as opposed to `defn`) used this way before. Is this just a convention, or is there a more in-depth reason for this?

Reply:
`defn` is a macro built on top of `def` and `fn`.

But here [`complement`](https://clojuredocs.org/clojure.core/complement) is a higher-order function that already a function so we just use `def`.
It's a sort of _point-free_ style of programming which I don't really like so use it very sparingly :).
An alternative would be: 
```
(defn atom? [exp]
  (not (list? exp)))
```